src/cacheExchange.ts

```diff
@@ -103,6 +122,23 @@ export const cacheExchange = (opts?: CacheExchangeOpts): Exchange => ({
     opts.keys
   );
 
+  let hydration;
+  if (opts.hydrate) {
+    hydration = opts.hydrate().then(({ connections, records, links }) => {
+      Object.keys(records).forEach(key => {
```

Comment:
I like the idea of saving objects and links but this can imply more sophisticated storage solution. 
With this approach, store will need to handle gc for the entire dataset and also have way to track links.
It will be much easier to save entire connections and objects as IndexDB objects (JSON.parse is not required for those). With that approach, we can simply ignore wiping out the objects (as they will be pruned already) etc.

See this comment (I intentionally split it to not link it to this repo:


https://github.com/apollographql/apollo-client/pull/
5601#issuecomment-556061183

Reply:
I think we’re just going to flatten the serialised structure even more, so the KV entries are going to be `conn:${connectionKey}`, `record:${recordKey}`, and `link:${linkKey}`. Our keys are already “hyper-normalised” in that all of these entries are already pretty granular. This will likely lend itself well to KV stores like IndexedDB.

I don’t think GC is in scope for the store itself. GC would have to be implemented on a Graphcache level, which we already started on prior to some refractors and then scrapped since we ran into some time constraints back then

Reply:
When working with a single key and json value in-store it is easy to save the entire object and also wipe data. In-memory implementation will probably need to truncate all entries before saving? Is it ok to leave some cache objects that are no longer needed/outdated

Reply:
This will eventually be solved with GC, but for now (for the minimum viable implementation) we will just write everything we have in batches to the storage (and we'll also add storing optimistic/offline mutations later on)